src/richie/apps/courses/models/category.py

```diff
@@ -117,6 +117,24 @@ def get_blogposts(self, language=None):
             .distinct()
         )
 
+    def get_children_categories(self, language=None):
+        """
+        Return a query to get only the children pages that are Category
+        objects related to this category. A convenient method since
+        ``page.get_child_pages`` return every direct child page, no matter
+        it's a Category or not.
+        """
+        page = (
+            self.extended_object
+            if self.extended_object.publisher_is_draft
+            else self.draft_extension.extended_object
```

Comment:
if you do this you will display the draft pages on the public page. I think you must just do:
```python
return self.extended_object.get_child_pages().filter(category__isnull=False)
```

Reply:
Changed as proposed